echo_handler.cc

```diff
@@ -4,11 +4,11 @@
 #include "request_handler.h"
 #include "echo_handler.h"
 
-std::string EchoHandler::GenerateResponse(const HttpParser& headerInfo, const std::string& requestData) const
+std::string EchoHandler::GenerateResponse(std::unique_ptr<Request> &request) const

```

Comment:
This doesn't have to be a unique_ptr. Changing this also breaks the API we're supposed to match. 

I recommend const Request&. The call can be changed like this: response = handler->GenerateResponse(*requestptr);